src/array/range.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { describe, expect, it } from 'vitest';
+import { range } from './range';
+
+describe('range', () => {
+  it('handles positive range with default step', () => {
+    expect(range(4)).toEqual([0, 1, 2, 3]);
+  });
+
+  it('handles negative range with default step', () => {
+    expect(range(-4)).toEqual([0, -1, -2, -3]);
+  });
+
+  it('handles custom positive step', () => {
+    expect(range(0, 20, 5)).toEqual([0, 5, 10, 15]);
+  });
+
+  it('handles custom negative step', () => {
+    expect(range(0, -4, -1)).toEqual([0, -1, -2, -3]);
+  });
+
+  it('handles zero step', () => {
+    expect(range(1, 4, 0)).toEqual([1, 1, 1]);
+  });
```

Comment:
Actually, I steps like `zero` should not permitted in `range` -- why not we throw an error here?

```tsx
throw new Error(`step must not be zero.`);
```

For example, Python's range function throws an error when `step` is zero.

<img width="325" alt="image" src="https://github.com/toss/es-toolkit/assets/3102175/74086224-1eca-4302-b1a7-ccf40895edc0">

Reply:
I think we should check if `step` is an integer (using `Number.isInteger`) and not zero..

Reply:
[lodash](https://lodash.com/docs/4.17.15#range) supports the step=0 case, and I think supporting the step=0 case would allow lodash users to migrate libraries to es-toolkit without breaking changes. what do you think? 
I guess it's a matter of choice, if you think it's better to follow the python implementation, I'll change it.

Reply:
As you mentioned, this will help people transition from lodash. However, I believe that prioritizing a better interface and consistent behavior is more important. 

It seems the behavior when the step is zero isn't well-defined, so I think we should choose not to support that use case.